src/hal/src/command/mod.rs

```diff
@@ -604,6 +604,46 @@ pub trait CommandBuffer<B: Backend>: fmt::Debug + Any + Send + Sync {
     /// Requests a timestamp to be written.
     unsafe fn write_timestamp(&mut self, stage: pso::PipelineStage, query: query::Query<B>);
 
+    /// TODO docs
+    /// `build_range_infos` must be the same length as `infos` and each element must have a length equal to the parallel `info` entry's `geometries.len()`. there's probably a way to make this safer without mangling the api shape too much...
+    unsafe fn build_acceleration_structures(
+        &self,
+        infos: &[acceleration_structure::AccelerationStructureGeometryDesc<B>],
+        build_range_infos: &[&[acceleration_structure::AccelerationStructureBuildRangeDesc]],
+    );
+
+    /// TODO docs
+    /// `indirect_device_addresses` is an array of infoCount buffer device addresses which point to infos[i]->geometryCount VkAccelerationStructureBuildRangeInfoKHR structures defining dynamic offsets to the addresses where geometry data is stored, as defined by infos[i].
+    unsafe fn build_acceleration_structures_indirect(
+        &self,
+        infos: &[acceleration_structure::AccelerationStructureGeometryDesc<B>],
+
+        indirect_device_addresses: &[(B::Buffer, buffer::Offset)],
+        indirect_strides: &[u32],
+        max_primitive_counts: &[&u32],
+    );
+
+    /// TODO docs
+    unsafe fn copy_acceleration_structure(
+        &self,
+        src: B::AccelerationStructure,
+        dst: B::AccelerationStructure,
+        mode: acceleration_structure::AccelerationStructureCopyMode,
+    );
+
+    // for host ops?
+    // - copy_acceleration_structure_to_memory
```

Comment:
names look fine to me!

Reply:
I initially had these commented out since these potentially depend on the [`VkDeviceOrHostAddressConstKHR`](https://www.khronos.org/registry/vulkan/specs/1.2-extensions/man/html/VkDeviceOrHostAddressConstKHR.html) type (which I've called `cpu-repr` in the comments), but I think I can support the GPU buffer variant of it for now